### PR TITLE
[receiver/dockerstats] Remove featuregate and old scraper implementation

### LINKED ISSUE #9794: [receiver/dockerstatsreceiver] Improvements proposal for dockerstatsreceiver
Looking to invest in the `dockerstatsreceiver`, and I noticed there are a couple of improvements that we could make to the component. I am happy to implement the below improvements once discussed here.

EDIT: This list below is the initial list. See my below comment for a refined list. 

1. Configuration ability to turn certain metrics on/off. Similar to `hostmetricsreceiver`. A set of default metrics will be on by default, and then the user has the ability to turn each metric on/off as they please. This is achieved by using `mdatagen`.
2. Add a new metric `memory.usage.total_cache`. This is something we need to get a better picture of the container's memory usage. This metric can be off-by-default assuming item (1) is implemented. 
3. Documentation improvements. This also can be achieved by onboarding the component to `mdatagen` since it auto generates the documentation for each metric. 
4. (Breaking change) Rename configuration options to use attributes, not labels. For example, `container_labels_to_metric_labels` would be renamed to `container_attributes_to_metric_attributes`
5. Once these items are implemented, declare the component as `alpha` with the aim to eventually move it into `beta`.

### PR DESCRIPTION
**Pull request by jamesmoessis**
**Description:** 
* Removes featuregate  that was marked stable last release. 
* Removes old implementation that featuregate toggled
* Removes deprecated config option `provide_per_core_cpu_metrics`
* There's no new code. Any green lines are things that I shuffled around so everything is in `receiver.go` (removed `metrics.go` and `receiver_v2.go`).

**Link to tracking Issue:** Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794

**Testing:** 
No behaviour changes. Same metrics are still produced, and this is validated by existing tests. 

**Documentation:** 
Updated README.

### CODE REVIEW

File: receiver/dockerstatsreceiver/receiver.go [hunk 1]
```diff
@@ -88,29 +92,210 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
...
+	}
+	return cpuPercent
+}
+
+func calculateMemoryPercent(memoryStats *dtypes.MemoryStats) float64 {
+	if float64(memoryStats.Limit) == 0 {
+		return 0
+	}
+	return 100.0 * (float64(memoryStats.Usage) - float64(memoryStats.Stats["cache"])) / float64(memoryStats.Limit)
 }
```
Comment (MovieStoreGuy) [inline receiver/dockerstatsreceiver/receiver.go:301]: Would it be worth moving to its own file named `thirdparty.go` just so it is clear on this is the only code copied and modified?

**Review by MovieStoreGuy — APPROVED** [review]
Changes look fine to me :) 

Just an outstanding question that @jpkrohling might be able to answer?

File: receiver/dockerstatsreceiver/receiver.go (hunk 1 above)
Comment (jamesmoessis) [reply receiver/dockerstatsreceiver/receiver.go:301]: This was just moved from `metrics.go` in this PR. Happy to move it elsewhere if required.

**Review by mx-psi — CHANGES_REQUESTED** [review]
Putting this on hold until we resolve the conversation over at #17981 (specifically see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17981#discussion_r1084727981 ), we may want to keep the feature flag definition in the code (even if we remove the old implementation)

**Comment by jamesmoessis** [comment]
@mx-psi - that conversation hasn't budged in two weeks now, I don't mind if we keep the feature flag definition either way. Do you want me to do that and I can remove it later depending on the outcome of that conversation?

**Comment by mx-psi** [comment]
> @mx-psi - that conversation hasn't budged in two weeks now. I'm happy to keep the feature flag definition to prevent errors. I can remove it later depending on the outcome of that conversation. Thoughts on that?

That makes sense to me, we can make this PR only remove the old implementation and merge it now, and keep the feature flag for later

File: receiver/dockerstatsreceiver/factory.go [hunk 2]
```diff
@@ -20,29 +20,17 @@ import (
...
 
 const (
-	typeStr        = "docker_stats"
-	stability      = component.StabilityLevelAlpha
-	useScraperV2ID = "receiver.dockerstats.useScraperV2"
+	typeStr   = "docker_stats"
+	stability = component.StabilityLevelAlpha
 )
 
-func init() {
```
Comment (mx-psi) [inline receiver/dockerstatsreceiver/factory.go:36]: so, to make my previous comment clear, I am fine with merging this PR if we don't remove this; everything else can be merged now IMO

**Comment by jamesmoessis** [comment]
Not sure what happened with the commit history there. I will close this and re open.

**Pull request closed without merging**